core/app/services/spree/checkout/add_store_credit.rb

```diff
@@ -0,0 +1,49 @@
+module Spree
+  module Checkout
+    class AddStoreCredit
+      prepend Spree::ServiceModule::Base
+
+      def call(order:, amount: nil)
+        @order = order
+        return failed unless @order
+
+        @order.payments.store_credits.where(state: :checkout).map(&:invalidate!)
+        remaining_total = amount ? [amount, @order.outstanding_balance].min : @order.outstanding_balance
+
+        apply_store_credits(remaining_total) if @order.user&.store_credits&.any?
+
+        @order.reload.payments.store_credits.valid.any? ? success(@order) : failure(@order)
+      end
+
+      private
+
+      def apply_store_credits(remaining_total)
+        payment_method = Spree::PaymentMethod::StoreCredit.available.first
+        raise 'Store credit payment method could not be found' unless payment_method
```

Comment:
@caudatecoder probably we need to change it to `failure` right?

Reply:
@damianlegawiec this part looks like a thing that should warn a developer in the dev environment. An exception here will be more detectable 
Though, it might be safer to raise only in dev and test environments. Like `ActiveRecord::PendingMigrationError`
What do you think?

Reply:
@caudatecoder ok, let's leave as it is for now